[DOCS-198] Add GitLab to existing CI/CD docs (#521)
algorithmiaio/dev-center: _pages/algorithm-development/ci-cd.md

> Continuous Integration & Deployment are standard practice in the world of software development, and Machine Learning is no exception—you need a robust CI/CD workflow to ensure that your latest models are deployed efficiently and correctly into production.

Reviewer: ```suggestion
Continuous Integration & Deployment are standard practice in the world of software development, and machine learning is no exception—you need a robust CI/CD workflow to ensure that your latest models are deployed efficiently and correctly into production.
```

> Algorithmia supports deployment and redeployment via the [the Algorithmia API]({{site.baseurl}}/algorithm-development/algorithm-management), and this is easily integrated into CI/CD tools such as Jenkins, GitHub Actions and GitLab CI/CD. With Algorithmia and your favorite CI/CD tool, your models are deployed as soon as they're ready, and they can be redeployed instantly whenever an approved retrained model is available.

Reviewer: ```suggestion
Algorithmia supports deployment and redeployment via the [the Algorithmia API]({{site.baseurl}}/algorithm-development/algorithm-management), and this is easily integrated into CI/CD tools such as Jenkins, GitHub Actions, and GitLab CI/CD. With Algorithmia and your favorite CI/CD tool, your models are deployed as soon as they're ready, and they can be redeployed instantly whenever an approved retrained model is available.
```

For more details on the configuration and capabilities of our GitHub Action, check out our detailed documentation on the GitHub Actions marketplace.
<a href="https://github.com/algorithmiaio/algorithmia-modeldeployment-action" class="btn btn-default btn-primary"><i class="fa fa-github" aria-hidden="true"></i> TRY IT OUT: GitHub Actions for Model Training Repo</a>

## GitLab CI/CD

> To deploy your algorithm files to Algorithmia everytime you push to your GitLab repository, make sure you have the following configured. 

Reviewer: ```suggestion
To deploy your algorithm files to Algorithmia every time you push to your GitLab repository, make sure you have the following configured. 
```

### Algorithm Files

> Similar to your default algorithm repo template, make sure your `src` folder and `requirements.txt` are in the root level of the repository. 

Reviewer: ```suggestion
Similar to your default algorithm-repository template, make sure your `src` folder and `requirements.txt` are in the root level of the repository. 
```

Reviewer: i think this is clearer, otherwise it sounds like "default" refers to "algorithm" not "algorithm repository template"

### Gitlab CI file and Algorithmia CI scripts
To get started with your pipelines, make sure you have the following at the root level of your repository:
- `.gitlab-ci.yml` file 
- `ci_entrypoint.py` file, as the entrypoint to your Algorithmia Deployment CI jobs

> - `algorithmia_ci` module directory, containing the scripts for deploying to Algorithmia and publishing a new version after runnning your defined tests.

Reviewer: ```suggestion
- `algorithmia_ci` module directory, containing the scripts for deploying to Algorithmia and publishing a new version after runnning your defined tests
```

### Gitlab CI Environment Variables

> Go to your repository's Settings -> CI/CD page and expand the Variables section. Add the following CI/CD variables so that your pipeline job scripts can access these values in a secured manner.

Reviewer: ```suggestion
Go to your repository's **Settings** -> **CI/CD** page and expand the **Variables** section. Add the following CI/CD variables so that your pipeline job scripts can access these values in a secure manner.
```

Reviewer: I've been **bolding** names like this so that they pop out when reading quickly through the docs.

Author: that's a good suggestion to keep in mind!

- `ALGO_NAME`: Algorithm name 
- `ALGO_USER`: Algorithmia username
- `ALGO_REPO_URL`: Repository host of the algorithm. Example: git.algorithmia.com
- `ALGO_PUBLISH_SCHEMA`: Semantic versioning of choice. Can be "major", "minor", or "revision"
- `API_ADDRESS`: The Algorithmia API cluster address to connect. Example: https://api.algorithmia.com

> - `API_KEY`: Algorithmia API key that has access to the configured algorithm and management capable. More information about Algorithmia API keys can be found [here](https://algorithmia.com/developers/platform/customizing-api-keys).

Reviewer: ```suggestion
- `API_KEY`: Algorithmia API key that has access to the configured algorithm and which is management capable. More information about Algorithmia API keys can be found [here](https://algorithmia.com/developers/platform/customizing-api-keys).
```

### Test Case Files
If you want to run tests before publishing a new version of your algorithm, you should add a new file called `TEST_CASES.json` at the root level of your repository. If you do not provide this file, then the CI/CD pipeline will omit the testing step.

> The required schema of this json file is shown below:

Reviewer: ```suggestion
The required schema of this JSON file is shown below.
```

#### Case Schema

> Your test cases should follow the following json schema

Reviewer: ```suggestion
Your test cases should follow the following JSON schema.
```

```
[
 { 
    "case_name": String,
    "input": Any,
    "expected_output": Any,
    "type": String,
    "tree": List
  },
  ...
]
```

- `input` (required) - the raw input that will be passed into the algorithm. Typically this will be a JSON dictionary, JSON list, or a primitive type (like a string).

> - `expected_output` (required) - What we are comparing against the result of your algorithm, which can be scoped in conjunction with setting `tree`. 

Reviewer: ```suggestion
- `expected_output` (required) - what we are comparing against the result of your algorithm, which can be scoped in conjunction with setting `tree`. 
```

For types `GREATER_OR_EQUAL` and `LESS_OR_EQUAL` this must be a number value. For types `NO_EXCEPTION` and `EXCEPTION` this field is optional.

> - `type` (optional) - defines the type of matching that can be done, options include `EXACT_MATCH`, `GREATER_OR_EQUAL`, `LESS_OR_EQUAL`, `NO_EXCEPTION` and `EXCEPTION`. Defaults to `EXACT_MATCH`

Reviewer: ```suggestion
- `type` (optional) - defines the type of matching that can be done. Options include `EXACT_MATCH`, `GREATER_OR_EQUAL`, `LESS_OR_EQUAL`, `NO_EXCEPTION`, and `EXCEPTION`. Defaults to `EXACT_MATCH`.
```

> - `tree` (optional) - A list defining the json keys we should traverse in order to find the value you wish to compare against with `expected_output`.

Reviewer: ```suggestion
- `tree` (optional) - a list defining the JSON keys we should traverse in order to find the value you wish to compare against with `expected_output`.
```
